src/game/TacticalAI/NPC.cc

```diff
@@ -1876,6 +1876,12 @@ void ConverseFull(UINT8 const ubNPC, UINT8 const ubMerc, Approach bApproach, UIN
 						}
 						else
 						{
+							// This check prevents NPCs from forever lingering if the player unloads the sector
+							// before an NPC chosen for removal reaches their destination (issue #750).
+							if (pNPC && pQuotePtr->sActionData == NPC_ACTION_REMOVE_NPC)
+							{
+								gMercProfiles[pNPC->ubProfile].sSector = SGPSector();
```

Comment:
Nice, I like bug fixes that are a net negative in code lines. I'm wondering why you need to check if pNPC is not null, though. ubNPC should always be the same as pNPC->ubProfile and gMercProfiles[ubNPC] is available under the variable name `p`.

Or as an alternative, wouldn't moving the line 					`gMercProfiles[ ubTargetNPC ].sSector = SGPSector();` in the NPC_ACTION_REMOVE_NPC case in `HandleNPCDoAction()` out of the `if ( pSoldier )` block also work?

Reply:
> wondering why you need to check if pNPC is not null

Just to be on the safe side. With `p` the check is not needed of course.

> gMercProfiles[ubNPC] is available under the variable name `p`

All the examples of using `SGPSector()` in the codebase are with `gMercProfiles[id]` so I didn't put much thought into it.

> wouldn't moving the line `gMercProfiles[ ubTargetNPC ].sSector = SGPSector();` in the NPC_ACTION_REMOVE_NPC case in `HandleNPCDoAction()` out of the `if ( pSoldier )` block also work?

Whatever happens in the NPC_ACTION_REMOVE_NPC case is irrelevant to this issue because it happens only after the destination is reached. And it's never reached if the map is unloaded during the travel. Spawning on the destination tile on sector load is not considered reaching.
And we can't do this action beforehand because it'll make the NPC disappear right in front the player's eyes because of `TacticalRemoveSoldier(*pSoldier)`.

Reply:
Ok then, thanks for the explanation.